specification/hdinsight/resource-manager/Microsoft.HDInsight/2015-03-01-preview/cluster.json

```diff
@@ -505,6 +451,13 @@
                     "$ref": "#/definitions/VirtualNetworkProfile",
                     "description": "The virtual network profile."
                 },
+                "dataDisksGroups": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/DataDisksGroups"
+                    },
+                    "description": "The data disks groups for the role."
+                },

```

Comment:
Addition of a new property without upgrading api-version is a breaking change.

Reply:
This is an optional property and for this specific case, there should be no compatibility issues with prior SDKs.

Reply:
It seems this model is used only for `Cluster_Create` operation 
which does a PUT on the endpoint `"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HDInsight/clusters/{clusterName}"`. 

Can customer use the same method to update a cluster? i mean can user get an existing Cluster instance, modify it locally and specify modified instance as a PUT payload (via `Cluster_Create` call)?

Reply:
A customer can't update their cluster with that same PUT call, e.g. if a customer creates a cluster with data disks, they cannot update it to remove the data disks. They would have to delete and create a separate cluster without data disks to do so.

Reply:
Fine, if that is the case then this change won't cause compatibility issues with prior SDK generated for the same api version.